datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/KafkaBrokerAddress.java

```diff
@@ -112,16 +104,6 @@ private static String validateHostname(String hostName) throws IllegalArgumentEx
     if (trimmed.isEmpty()) {
       throw new IllegalArgumentException("empty host name");
     }
-    // we allow ipv4/6 addresses and RFC 1123 host names
```

Comment:
If we remove this, what would be the final error we will see if the address format isn't correct by accident for a typical "confirming" address? Would it be just generic send error failures? If that's the case, instead of removing the check altogether, if possible, might want to add a setting to relax the check . Esp. if majority of the broker addresses are going to be confirming types, as this gives us a clear message about what's wrong.

Reply:
I agree. 

However, the problem is that he call chain is as follows:
`AbstractKafkaConnector`
  → `KafkaConnectionString::valueOf()`
  → `KafkaConnectionString::parseBrokers()`
  → `KafkaBrokerAddress::valueOf()`
  → `KafkaBrokerAddress::validateHostname()`

If we introduce a flag to toggle hostname validation, we'd have to wire it through this whole chain.

Reply:
@gauravrkulkarni 
Another key thing I forgot to mention: removing this validation has not impact on reliability because even if a hostname is IPv4/6 or RFC 1123 compliant, there's no guarantee that it exists or is resolvable through DNS. Even with this validation, we can still create datastreams against perfectly valid hostnames that simply do not exist. However, we cannot create datastreams against hostnames that do exist just because their names aren't IPv4/6 or RFC 1123 compliant.

Reply:
If its too much of a hassle to add a config and if the merits aren't that many, its ok to not have config.